### ISSUE TITLE
Missing check for compact method name

### ISSUE THREAD

**Issue by ericelliott** [johnstonbl01/eslint-no-inferred-method-name#1]
The real issue is with compact object literals. Here's the example we need to fix:

```
const repeater = {
  name: 'Repeater',
  types: [
    { f: 'function' },
    { n: 'number' }
  ],
  repeat (f, n) {
    let i = n - 1;

    if (typeof f === 'function') {
      f();
    } else {
      throw new Error('repeat: A Function is required.');
    }
    if (!n) {
      return undefined;
    }

    return repeat(f, i);
  }
};

repeater.repeat(() => {console.log('foo');}, 10);
```

Results:

```
foo
ReferenceError: repeat is not defined
```

The way to fix that is to report an error when the method name is used within the function, because inside the function, the method name is undefined. The error should be reported for this line:

```
    return repeat(f, i);
```

The built-in `no-undef` rule already catches this error, but we need to explicitly tell the user that compact method names are not available inside the method in order to clear up any confusion.

**Comment by ericelliott** [comment]
In other words, we need to add a couple tests:

```
valid: [
  { code: "var obj = { foo() { } }"}
],
invalid: [
  { code: "var obj = { foo() { foo(); } }"}
],
```

**Comment by johnstonbl01** [comment]
I've integrated these tests in the latest commit. No changes to the actual rule though. All the tests pass, but let me know if I'm missing something.

I run `eslint` on a a file I placed the code above in and get:

```
sample.js
   1:0   error    Strings must use doublequote                                              quotes
   5:8   error    Strings must use doublequote                                              quotes
   7:9   error    Strings must use doublequote                                              quotes
   8:9   error    Strings must use doublequote                                              quotes
  13:21  error    Strings must use doublequote                                              quotes
  16:22  error    Strings must use doublequote                                              quotes
  22:11  warning  "repeat" is not defined                                                   no-undef
  22:11  error    "repeat" has no defined method name. Use syntax - foo: function foo {..}  no-inferred-method-name
  26:35  error    Strings must use doublequote                                              quotes
  26:41  error    Missing whitespace after semicolon                                        semi-spacing

✖ 10 problems (9 errors, 1 warning)
```

As you can see on line 22:11, it's flagging an error on the function call inside the compact object literal. No-undef will still flag an issue or warning for the line in addition to the custom rule. Again, let me know if I've missed something!

**Comment by ericelliott** [comment]
> As you can see on line 22:11, it's flagging an error on the function call inside the compact object literal. No-undef will still flag an issue or warning for the line in addition to the custom rule. Again, let me know if I've missed something!

Rocked it!

:100:

**Comment by johnstonbl01** [comment]
Thanks, man!

**Comment by ericelliott** [comment]
You want to add an example of the compact literal to the README?

**Comment by johnstonbl01** [comment]
Yes, definitely will do.

**Comment by ericelliott** [comment]
Great! =)

**Comment by johnstonbl01** [comment]
Done!
